terraform/variables/README.md

```diff
@@ -0,0 +1,205 @@
+
+- A TFC organization name. This is arbitrary, but needs to be globally unique.
+
+- An email address to use as your TFC org admin contact address.
+
+Run Terraform to create the TFC org and workspace. You'll be prompted for the above inputs
+
+```shell
+$ cd ./terraform/tfc
+$ terraform init
+...
+$ terraform apply
+...
+Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
+```
+
+### Generating Terraform Outputs
+
+In `terraform/main`, we have the minimal possible Terraform to generate ouputs. This is meant to simulate
+real Terraform that might be creating and AWS VPC, and outputting things like VPC IDs and subnet arns.
+
+You'll need the following inputs:
+
+- An org ID and workspace ID. These cannot be referenced as variables, so you'll need to modify 
+`./terraform/main/main.tf`. Use the organization name chosen in the previous step, or set this
+to your own organization ID if desired.
+
+```hcl
+terraform {
+  cloud {
+    organization = "waypoint_tfc_vars_example" // <- you must change this
+
+    workspaces {
+      name = "waypoint_tfc_vars_example"
+    }
+  }
+}
+```
+
+Run the Terraform to produce the outputs:
+```shell
+$ cd ./terraform/main
+$ terraform init
+...
+$ terraform apply
+...
+Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+list_of_strings = [
+  "item_1",
+  "item_2",
+  "item_3",
+]
+map_of_strings = {
+  "key1" = "value1"
+  "key2" = "value2"
+}
+simple_string = "justastring"
+```
+
+## Waypoint setup
+
+#### Waypoint CLI
+
+Install the Waypoint CLI. See [installation instructions](https://developer.hashicorp.com/waypoint/tutorials/get-started-docker/get-started-install).
+
+#### Waypoint Server
+
+You'll need a Waypoint server. We recommend using [HCP Waypoint](https://cloud.hashicorp.com/products/waypoint)
+You'll also need your CLI configured with a [context](https://developer.hashicorp.com/waypoint/commands/context-create)
+pointing to this server.
+
+#### Waypoint remote runner
+
+Waypoint remote runners are required to use variables with dynamic defaults. HCP Waypoint will 
+walk you through installing a remote runner into your infrastructure. If you don't have
+a runner already, you can run `waypoint runner agent` in a separate terminal window. This
+runner will have access to docker on your laptop, which is where we'll be deploying.
+
+#### Waypoint Terraform Configsourcer Config
+
+Waypoint needs to know how to talk to Terraform cloud at runtime to source the outputs. 
+To accomplish this, run:
+
+```shell
+waypoint config source-set -type=terraform-cloud -config=token=<your-tfc-token>
+```
+#### Customize the waypoint.hcl
+
+
+Take a look at the waypoint.hcl. In each variable stanza, you'll need to update the
+Terraform organization to match the one your chose during the Terraform setup
+
+Example
+```hcl
+variable "single_string_output" {
+  default = dynamic("terraform-cloud", {
+    organization = "waypoint_tfc_vars_example" // REPLACE WITH YOUR ORG ID
```

Comment:
I wonder how hard it would be to add logic for tf vars to be reference-able from the waypoint.hcl since both are .hcl vars. So if that could work it would simplify hard coding the org to just be something like `var.tfc_org_name` from the `variables.tf` or grab it from the resource in the `main.tf` itself by using something like `tf_resource. tfe_organization.tfe_organization`. Just food for thought :)

Reply:
That's a good idea! Will keep that in the back of my mind as a future improvement 😄